packages/react/src/form/Form.tsx

```diff
@@ -17,13 +17,21 @@ const Form = React.forwardRef(function Form(
   props: Form.Props,
   forwardedRef: React.ForwardedRef<HTMLFormElement>,
 ) {
-  const { render, className, errors, onClearErrors: onClearErrorsProp, ...otherProps } = props;
+  const {
+    render,
+    className,
+    errors,
+    onClearErrors: onClearErrorsProp,
+    onSubmit: onSubmitProp,
+    ...otherProps
+  } = props;
 
   const formRef = React.useRef<FormContext['formRef']['current']>({
     fields: new Map(),
   });
   const submittedRef = React.useRef(false);
 
+  const onSubmit = useEventCallback(onSubmitProp || (() => {}));
```

Comment:
nit: we could define a NOOP constant and reuse it (also in onClearErrors below) instead of defining a new function on each render.

Reply:
I'd prefer if `useEventCallback` handled nil values actually

Reply:
Yeah, that's a good approach.